packages/backend/src/services/RenameService/rename.ts

```diff
@@ -0,0 +1,823 @@
+    metrics: filters.metrics
+        ? (renameFilterGroups(
+              filters.metrics,
+              replaceModelPrefix,
+          ) as FilterGroup)
+        : undefined,
+    tableCalculations: filters.tableCalculations
+        ? (renameFilterGroups(
+              filters.tableCalculations,
+              replaceModelPrefix,
+          ) as FilterGroup)
+        : undefined,
+});
+
+export const renameDashboardFilterRules = (
+    filterRules: DashboardFilterRule[],
+    { replaceFull, replaceId }: ReturnType<typeof createRenameFactory>,
+): DashboardFilterRule[] =>
+    filterRules.map((filterRule) => ({
+        ...filterRule,
+        target: {
+            ...filterRule.target,
+            fieldId: replaceId(filterRule.target.fieldId),
+            tableName: replaceFull(filterRule.target.tableName),
+        },
+        tileTargets: filterRule.tileTargets
+            ? Object.fromEntries(
+                  Object.entries(filterRule.tileTargets).map(([key, value]) => [
+                      key,
+                      value
+                          ? {
+                                ...value,
+                                fieldId: replaceId(value.fieldId),
+                                tableName: replaceFull(value.tableName),
+                            }
+                          : value,
+                  ]),
+              )
+            : undefined,
+    }));
+
+const renameDashboardFilters = (
+    filters: DashboardFilters,
+    methods: ReturnType<typeof createRenameFactory>,
+): DashboardFilters => ({
+    dimensions: renameDashboardFilterRules(filters.dimensions, methods),
+    metrics: renameDashboardFilterRules(filters.metrics, methods),
+    tableCalculations: renameDashboardFilterRules(
+        filters.tableCalculations,
+        methods,
+    ),
+});
+
+export const renameMetricQuery = (
+    metricQuery: MetricQuery,
+    {
+        replaceId,
+        replaceList,
+        replaceFull,
+        replaceReference,
+        replaceFieldReference,
+        replaceFieldName,
+    }: ReturnType<typeof createRenameFactory>,
+): MetricQuery => {
+    const updatedMetricQuery = {
+        ...metricQuery,
+        exploreName: replaceFull(metricQuery.exploreName),
+        dimensions: replaceList(metricQuery.dimensions),
+        metrics: replaceList(metricQuery.metrics),
+        filters: renameFilters(metricQuery.filters, replaceId),
+        tableCalculations: metricQuery.tableCalculations?.map((tc) => ({
+            ...tc,
+            sql: replaceReference(tc.sql),
+        })),
+        additionalMetrics: metricQuery.additionalMetrics?.map((am) => ({
+            ...am,
+            table: replaceFull(am.table),
+            baseDimensionName: am.baseDimensionName
+                ? replaceFieldName(am.baseDimensionName)
+                : undefined,
+            sql: replaceFieldReference(am.sql), // This reference can contain a ${TABLE} too
+        })), // TODO replace also filters
```

Comment:
I think this is done above and comment can be removed

Reply:
this is for custom metric filters